flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -875,7 +875,8 @@ public final void cleanUp(Throwable throwable) throws Exception {
 
         failing = !canceled && (!(throwable instanceof CancelTaskException));
 
-        Exception suppressedException = runAndSuppressThrowable(this::cancelTask, null);
+        Exception suppressedException =
+                throwable == null ? null : runAndSuppressThrowable(this::cancelTask, null);
```

Comment:
If you pass `supressedException` to `runAndSuppressThrowable` you can avoid catching exceptions when you call `cleanUp`. I mean it will be possible to transform code `restoreAndInvoke` from:
```
try {
           ....
} catch (Throwable throwable) {
    try {
        runWithSystemExitMonitoring(() -> finalInvokable.cleanUp(throwable));
    } catch (Throwable cleanUpThrowable) {
        throwable.addSuppressed(cleanUpThrowable);
    }
    throw throwable;
}
```
to 
```
try {
           ....
} catch (Throwable throwable) {
    runWithSystemExitMonitoring(() -> finalInvokable.cleanUp(throwable));
}
```

Reply:
I decided not to suppress throwable in `cleanUp` because:
1. There can be other implementations that do throw an exception from `cleanUp`
2. `runAndSuppressThrowable` returns an `Exception`, not `Throwable`; so it would either wrap it as it does now, or `cleanUp` should declare that it throws `Throwable`; both of these are undesirable IMO